File: Easy/1636. Sort Array by Increasing Frequency/frequencySort.cpp
class Solution {
public:
    vector<int> frequencySort(vector<int>& nums) {
        /*
            先用MAP紀錄每一個數字出現的次數
            再去查這個表格來排序
            Time complexity: O(n*log(n))
            Space complexity: O(n)
        */
        unordered_map<int, int> fq;
        for(int n : nums){
            fq[n]++;
        }
        sort(nums.begin(), nums.end(), [&](int A, int B){
            if(fq[A] == fq[B]){
                return A > B;
            }
            return fq[A] < fq[B];
        });
        return nums;
    }
};

File: Easy/1636. Sort Array by Increasing Frequency/frequencySort.java
class Solution {
    public int[] frequencySort(int[] nums) {
        /*
            記住java如何寫comparator
        */
        Map<Integer, Integer> m = new HashMap<>();
        for(int i = 0; i < nums.length; i++){
            m.put(nums[i], m.getOrDefault(nums[i], 0) + 1);
        }
        Integer [] sorted = new Integer[nums.length];
        for(int i = 0; i < nums.length; i++){
            sorted[i] = nums[i];
        }
        Arrays.sort(sorted, new Comparator<Integer>(){
            @Override
            public int compare(Integer a, Integer b){
                if(m.get(a) == m.get(b)){
                    return b - a;
                }
                return m.get(a) - m.get(b);
            }
        });
        for(int i = 0 ; i < nums.length; i++){
            nums[i] = sorted[i];
        }
        return nums;
    }
}